google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/ExtractJobConfiguration.java

```diff
@@ -296,8 +318,15 @@ public int hashCode() {
 
   @Override
   ExtractJobConfiguration setProjectId(String projectId) {
-    if (Strings.isNullOrEmpty(getSourceTable().getProject())) {
-      return toBuilder().setSourceTable(getSourceTable().setProjectId(projectId)).build();
+    if (getSourceTable() != null) {
+      if (Strings.isNullOrEmpty(getSourceTable().getProject())) {
+        return toBuilder().setSourceTable(getSourceTable().setProjectId(projectId)).build();
+      }
+    }
+    if (getSourceModel() != null) {
+      if (Strings.isNullOrEmpty(getSourceModel().getProject())) {
+        return toBuilder().setSourceModel(getSourceModel().setProjectId(projectId)).build();
+      }
```

Comment:
I think combining two if block would be good idea

```java
if (getSourceTable() != null && isNullOrEmpty(getSourceTable().getProject())) {
   return toBuilder().setSourceTable(getSourceTable().setProjectId(projectId)).build();
}
```

similarly for `getSourceModel()`

Reply:
I have tried that idea before  if getSourceTable() is null and you are checking getProject() but actually it's null  it will throw an error `NullPointerException`.